### PR TITLE
fix: Remove the onClose subscribers notifications and notifies to the normal subscribers before removing the wallet

### PR DESCRIPTION
**Pull request by banklesss**
There was issues with the wallet subscriptions when the wallet closes and subscriptors are notified. 

That subscription has been refactored and those actions are performed every time a wallet is closed.

### CODE REVIEW

**Review by stackchain — APPROVED** [review]

**Comment by stackchain** [comment]
@jemubm I've added a few more clean-ups.

**Comment by banklesss** [comment]
> @jemubm I've added a few more clean-ups.

Thanks !

File: src/hooks/index.ts [hunk 1]
```diff
@@ -470,6 +487,26 @@ export const useRemoveWallet = (options: UseMutationOptions<void, Error, void>)
   }
 }
 
+export const useResyncWallet = ({onSuccess, ...options}: UseMutationOptions<void, Error, void> = {}) => {
```
Comment (wolverineks) [inline src/hooks/index.ts:490]: is this used anywhere?

File: src/hooks/index.ts [hunk 2]
```diff
@@ -33,9 +36,16 @@ import {generateShelleyPlateFromKey} from '../yoroi-wallets/cardano/shelley/plat
...
 
 // WALLET
-export const useCloseWallet = (options?: UseMutationOptions<void, Error>) => {
+export const useCloseWallet = ({onSuccess, ...options}: UseMutationOptions<void, Error> = {}) => {
+  const dispatch = useDispatch()
+
   const mutation = useMutation({
     mutationFn: () => walletManager.closeWallet(),
+    onSuccess: (data, variables, context) => {
+      dispatch(clearUTXOs())
```
Comment (wolverineks) [inline src/hooks/index.ts:45]: dont add dispatch to these hooks

File: src/hooks/index.ts [hunk 3]
```diff
@@ -457,9 +467,16 @@ export const useWalletMetas = <T = Array<WalletMeta>>(options?: UseQueryOptions<
...
 }
 
-export const useRemoveWallet = (options: UseMutationOptions<void, Error, void>) => {
+export const useRemoveWallet = ({onSuccess, ...options}: UseMutationOptions<void, Error, void> = {}) => {
+  const dispatch = useDispatch()
+
   const mutation = useMutationWithInvalidations({
     mutationFn: () => walletManager.removeCurrentWallet(),
+    onSuccess: (data, variables, context) => {
+      dispatch(clearUTXOs())
```
Comment (wolverineks) [inline src/hooks/index.ts:476]: dont add dispatch to these hooks

File: src/hooks/index.ts (hunk 1 above)
Comment (banklesss) [reply src/hooks/index.ts:490]: applied here -> https://github.com/Emurgo/yoroi-mobile/pull/2125

File: src/hooks/index.ts (hunk 2 above)
Comment (banklesss) [reply src/hooks/index.ts:45]: applied here -> https://github.com/Emurgo/yoroi-mobile/pull/2125

File: src/hooks/index.ts (hunk 3 above)
Comment (banklesss) [reply src/hooks/index.ts:476]: applied here -> https://github.com/Emurgo/yoroi-mobile/pull/2125

**Pull request merged**